### PR TITLE
solution: 0101-symmetric-tree updated with java solution

### PR DESCRIPTION
**Pull request by vigneshshiv**
solution: 0101-symmetric-tree updated with java solution

### CODE REVIEW

File: solutions/0100-0199/0101-symmetric-tree-easy.md [hunk 1]
```diff
@@ -62,4 +64,119 @@ public:
...
+ *     TreeNode(int val) { this.val = val; }
+ *     TreeNode(int val, TreeNode left, TreeNode right) {
+ *         this.val = val;
+ *         this.left = left;
+ *         this.right = right;
+ *     }
+ * }
+ */
+// Time complexity: O(n), where n - # nodes in the tree
+// Space complexity: O(log(n))
```
Comment (wingkwong) [inline solutions/0100-0199/0101-symmetric-tree-easy.md:91]: why logn?

Comment (vigneshshiv) [reply solutions/0100-0199/0101-symmetric-tree-easy.md:91]: We can simply say space complexity O(h), h - height of the tree. 

OR

Suppose, we have perfectly balanced binary tree, we start with root that N, then next layer we have left & right subtrees so we have N/2, then next layer is N/4 so on till last. 

Roughly the pattern is

$(N)+(N/2)+(N/4)+(N/8)+... + (N/N)$ which is $O(logN)$ same as height of the tree.

Comment (wingkwong) [reply solutions/0100-0199/0101-symmetric-tree-easy.md:91]: I mean state the reason there :D

Comment (vigneshshiv) [reply solutions/0100-0199/0101-symmetric-tree-easy.md:91]: Sure.

**Comment by vigneshshiv** [comment]
Space complexity updated, please check

**Pull request merged**